server/database.sql

```diff
@@ -1,150 +1,378 @@
+  `id` int NOT NULL AUTO_INCREMENT,
+  `question_id` int NOT NULL,
+  `contest_id` int NOT NULL,
+  `max_score` int NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_contests_questions_1_idx` (`contest_id`),
+  KEY `fk_contests_questions_2_idx` (`question_id`),
+  CONSTRAINT `fk_contests_questions_1` FOREIGN KEY (`contest_id`) REFERENCES `contests` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_contests_questions_2` FOREIGN KEY (`question_id`) REFERENCES `questions` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `groups`
+--
+
+DROP TABLE IF EXISTS `groups`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `groups` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `group_name` varchar(45) NOT NULL,
+  `confidential` tinyint NOT NULL,
+  `creator` varchar(45) NOT NULL,
+  `member_count` int NOT NULL,
   PRIMARY KEY (`id`),
   KEY `groups_fk_creator` (`creator`),
-  CONSTRAINT `groups_fk_creator` 
-    FOREIGN KEY (`creator`) 
-    REFERENCES `users` (`username`) 
-    ON DELETE CASCADE 
-    ON UPDATE CASCADE) 
-    ENGINE = InnoDB 
-    DEFAULT CHARACTER SET = utf8mb4 
-    COLLATE = utf8mb4_0900_ai_ci;
-
-
--- -----------------------------------------------------
--- Table `notifications`
--- -----------------------------------------------------
-DROP TABLE IF EXISTS `notifications` ;
-
-CREATE TABLE IF NOT EXISTS `notifications` (
-  `id` INT NOT NULL AUTO_INCREMENT,
-  `heading` VARCHAR(110) NOT NULL,
-  `description` VARCHAR(255) NULL DEFAULT NULL,
-  `created_at` DATETIME NULL DEFAULT CURRENT_TIMESTAMP,
-  `creator` VARCHAR(45) NOT NULL,
+  CONSTRAINT `groups_fk_creator` FOREIGN KEY (`creator`) REFERENCES `users` (`username`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `leaderboard`
+--
+
+DROP TABLE IF EXISTS `leaderboard`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `leaderboard` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `rank` int NOT NULL,
+  `username` varchar(45) NOT NULL,
+  `contest_id` int NOT NULL,
+  `score` int NOT NULL,
+  `total_time` timestamp NOT NULL,
+  `attempted_count` int NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_leaderboard_users_idx` (`username`),
+  KEY `fk_leaderboard_contests_idx` (`contest_id`),
+  CONSTRAINT `fk_leaderboard_contests` FOREIGN KEY (`contest_id`) REFERENCES `contests` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_leaderboard_users` FOREIGN KEY (`username`) REFERENCES `users` (`username`) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
Similarly, 1 more constraint needs to be added to avoid unnecessary duplicate entries -
```
UNIQUE KEY `unique_contests_username` (`contest_id`,`username`)
```